### PR TITLE
[hail] Add RunAggScan IR node

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/ir/EmitStream.scala [hunk 1]
```diff
@@ -784,6 +784,32 @@ object EmitStream {
...
+          val bodyEnv = env.bind(name -> ((typeToTypeInfo(producerElementPType), eltm, eltv)))
+          val cInit = emitter.emit(init, env, er, Some(newContainer))
+          val seqPerElt = emitter.emit(seqs, bodyEnv, er, Some(newContainer))
+          val postt = emitter.emit(result, bodyEnv, er, Some(newContainer))
+
+          emitStream(array, env)
+            .contMap[EmitTriplet] { (eltt, k) =>
+              Code(
+                TypedTriplet(producerElementPType, eltt).storeTo(eltm, eltv),
+                TypedTriplet(resultPType, postt).storeTo(postm, postv),
```
Comment (patrick-schultz) [inline hail/src/main/scala/is/hail/expr/ir/EmitStream.scala]: It doesn't look like the `postm`, `postv` fields are necessary. Couldn't the returned emit triplet just forward the `m` and `v` from `postt`?

File: hail/src/main/scala/is/hail/expr/ir/EmitStream.scala [hunk 2]
```diff
@@ -784,6 +784,32 @@ object EmitStream {
...
+                seqPerElt.setup,
+                k(EmitTriplet(Code._empty, postm, postv)))
```
[the remark this answers is not in the document]
Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/EmitStream.scala]: I just copied the ArrayAggScan emitter. This is better, though, I think.

File: hail/src/main/scala/is/hail/expr/ir/EmitStream.scala (hunk 1 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/EmitStream.scala]: doesn't work, generates invalid bytecode. Don't want to dig into that now.

Comment (patrick-schultz) [reply hail/src/main/scala/is/hail/expr/ir/EmitStream.scala]: Huh, okay.

**Pull request merged**